flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperator.java

```diff
@@ -118,7 +118,10 @@ public AsyncWaitOperator(
 			int capacity,
 			AsyncDataStream.OutputMode outputMode) {
 		super(asyncFunction);
-		chainingStrategy = ChainingStrategy.ALWAYS;
+
+		// TODO this is a temporary fix for the problems described under FLINK-13063 at the cost of never chaining
+		//  AsyncOperators.
+		setChainingStrategy(ChainingStrategy.NEVER);
```

Comment:
Can we make this `HEAD`? I think there are no known issues that are forcing the `AsyncWaitOperator` to be also a tail of an operator chain?

Reply:
Ok, I think that is possible and was just a communication issue about the requirements.